src/world.fnl

```diff
@@ -54,6 +54,25 @@
 
 ; PRIVATE UTILS
 
+(fn collision-start-cb [f1 f2 contact]
+    (let [o1 (f1:getUserData)
+          o2 (f2:getUserData)]
+        (when (and o1 o2)
+            (o1:collide-with o2 contact)
+        )
+    )
+)
+
+(fn collision-end-cb [f1 f2 contact]
+    (let [o1 (f1:getUserData)
+          o2 (f2:getUserData)]
+        (when (and o1 o2)
+            (o1:part-with o2 contact)
+        )
+    )
+)
```

Comment:
These only call the `collide-with` and `part-with` callbacks if both objects have called `setUserData` on their `Fixture` instance. Otherwise the collision is ignored.